### PR TITLE
feat(editor): Setup Sentry integration

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This PR
1. Decouples the session started event from the settings endpoint
2. Sets up Sentry Vue SDK to report errors to Sentry, with `release` and `enviroment` as additional data

## Related Linear tickets, Github issues, and Community forum posts

[CAT-8](https://linear.app/n8n/issue/CAT-8)

## Review / Merge checklist

- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/editor-ui/src/shims.d.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ declare global {
 	interface Window {
 		BASE_PATH: string;
 		REST_ENDPOINT: string;
+		__n8n_settings: FrontendSettings;
```
Comment (alexgrozav) [inline packages/editor-ui/src/shims.d.ts:21]: nit: I feel like `__n8n_settings` is a bit pythonic. I'd personally prefer a plain, clear camelCase name like `n8nSettings`.

File: packages/@n8n/config/src/configs/sentry.config.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+@Config
+export class SentryConfig {
+	/** Sentry DSN for the backend */
+	@Env('N8N_SENTRY_DSN')
+	backend_dsn: string = '';
+
+	/** Sentry DSN for the frontend . */
+	@Env('N8N_FRONTEND_SENTRY_DSN')
+	frontend_dsn: string = '';
+}
```
Comment (ivov) [inline packages/@n8n/config/src/configs/sentry.config.ts:12]: Nit

```suggestion
	/** Sentry DSN for the backend. */
	@Env('N8N_SENTRY_DSN')
	backendDsn: string = '';

	/** Sentry DSN for the frontend. */
	@Env('N8N_FRONTEND_SENTRY_DSN')
	frontendDsn: string = '';
}
```

File: packages/cli/src/events/events.controller.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { AuthenticatedRequest } from '@/requests';
+
+import { EventService } from './event.service';
+
+@RestController('/events')
+export class EventsController {
+	constructor(private readonly eventService: EventService) {}
+
+	@Get('/session-started')
+	track(req: AuthenticatedRequest) {
```
Comment (ivov) [inline packages/cli/src/events/events.controller.ts:11]:
```suggestion
	sessionStarted(req: AuthenticatedRequest) {
```

File: packages/cli/src/events/events.controller.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+		const pushRef = req.headers['push-ref'] as string;
```
Comment (ivov) [inline packages/cli/src/events/events.controller.ts:12]: Every authenticated request has this push ref? If so let's reflect this in the type?

```ts
export type AuthenticatedRequest<
	RouteParams = {},
	ResponseBody = {},
	RequestBody = {},
	RequestQuery = {},
> = Omit<APIRequest<RouteParams, ResponseBody, RequestBody, RequestQuery>, 'user' | 'cookies'> & {
	user: User;
	cookies: Record<string, string | undefined>;
	headers: express.Request['headers'] & {
		'push-ref': string;
	};
};
```

File: packages/cli/src/events/events.controller.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Get, RestController } from '@/decorators';
+import { AuthenticatedRequest } from '@/requests';
+
+import { EventService } from './event.service';
+
+@RestController('/events')
```
Comment (ivov) [inline packages/cli/src/events/events.controller.ts:6]: Docline? People might confuse this with the event bus controller.

File: packages/editor-ui/index.html [hunk 6]
```diff
@@ -9,6 +9,7 @@
 			window.BASE_PATH = '/{{BASE_PATH}}/';
 			window.REST_ENDPOINT = '{{REST_ENDPOINT}}';
 		</script>
+		<script src="/{{REST_ENDPOINT}}/sentry.js"></script>
```
Comment (ivov) [inline packages/editor-ui/index.html:12]: So request the Sentry config even when Sentry is disabled?

File: patches/@sentry__cli@2.36.2.patch [hunk 7]
```diff
@@ -1,5 +1,5 @@
 diff --git a/js/helper.js b/js/helper.js
-index a4a7a61f0e226d7cb45f8d5db34c35cc9e62cf96..38fd3ec2214970b1c5dd22fdff902ff3b5eeddde 100644
+index 37798b9444d39a8713327ed12adf3e76f03188a4..08d49b7a5c058c4d29c9929c0c524c85b14b330e 100644
```
Comment (ivov) [inline patches/@sentry__cli@2.36.2.patch:2]: For learning, why did this change?

File: packages/editor-ui/index.html (hunk 6 above)
Comment (netroy) [reply packages/editor-ui/index.html:12]: yeah. frontend build doesn't know about if sentry is enabled or disabled. When we release, the frontend code is compiled into a bunch of static files, that don't change according to any runtime configuration (except N8N_PATH).
So, this script needs to be injected here to be able to get this dynamic information into the frontend before anything else loads.

File: patches/@sentry__cli@2.36.2.patch (hunk 7 above)
Comment (netroy) [reply patches/@sentry__cli@2.36.2.patch:2]: it's because of the `@sentry/cli` upgrade via the `@sentry/vite-plugin` upgrade.

**Review by ivov — APPROVED** [review]
🎉

**Comment by janober** [comment]
Got released with [`n8n@1.61.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.61.0)

**Pull request merged**